### ISSUE TITLE
small geographical area use case

### ISSUE THREAD

**Issue by radius314** [radius314/yap#11]
Martha's Vineyard area is doing something very interesting.  They have a small number of meetings (an isolated area), and only want to surface these meetings.   

The idea is to make a small entry point with minimal functionality.  Here is the prototype:

```php
<?php

	// set time zone
	date_default_timezone_set("America/New_York");

	header("content-type: text/xml");
    echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"; 
	
	// BML Search parameters, can easily be altered by going to semantic workshop and selecting json as output
	// semantic workshop url wp.nerna.org/main_server/semantic
	$bmlt_search_endpoint = file_get_contents($root . "/client_interface/json/?switcher=GetSearchResults&services=8&weekdays[]=1&weekdays[]=2&weekdays[]=3&weekdays[]=4&weekdays[]=5&weekdays[]=6&weekdays[]=7");

	// change days from numbers to text name
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"1"', 'weekday_tinyint":"Sunday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"2"', 'weekday_tinyint":"Monday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"3"', 'weekday_tinyint":"Tuesday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"4"', 'weekday_tinyint":"Wednesday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"5"', 'weekday_tinyint":"Thursday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"6"', 'weekday_tinyint":"Friday"', $bmlt_search_endpoint);
	$bmlt_search_endpoint = str_replace('weekday_tinyint":"7"', 'weekday_tinyint":"Saturday"', $bmlt_search_endpoint);

	// decodes the json results into object
	$search_results = json_decode($bmlt_search_endpoint);

	// sort starting with current day
	usort($search_results, function($a, $b){
    	return strtotime($a->weekday_tinyint) - strtotime($b->weekday_tinyint);
	});

    $results_count = 7;  //how many results to return
    $text_space = "\r\n";
    $message = "";
?>
<Response>
<?php 
    if (count($search_results) == 0) {
        echo "<Say voice=\"woman\">There are no other meetings for today.  Thank you for calling, goodbye.</Say>";
    } else {
?>
    <Say voice="woman">Marthas Vineyard Meeting information found, listing the top <?php echo $results_count ?> results.</Say>
<?php
        $results_counter = 0;
        for ($i = 0; $i < count($search_results); $i++) {  
            $result_day = $search_results[$i]->weekday_tinyint;
            $result_time = $search_results[$i]->start_time;
			
            $part_1 = str_replace("&", "&amp;", $search_results[$i]->meeting_name);
				
            $part_2 = str_replace("&", "&amp;", $result_day
                    . ' ' . (new DateTime($result_time))->format('g:i A'));
			
            $part_3 = str_replace("&", "&amp;", $search_results[$i]->location_street 
                    . " in " . $search_results[$i]->location_municipality 
                    . ", " . $search_results[$i]->location_province);

            echo "<Pause length=\"1\"/>";
            echo "<Say voice=\"woman\">Result number " . ($results_counter + 1) . "</Say>";
            echo "<Say voice=\"woman\">" . $part_1 . "</Say>";
            echo "<Pause length=\"1\"/>";
            echo "<Say voice=\"woman\">Starts at " . $part_2 . "</Say>";
            echo "<Pause length=\"1\"/>";
            echo "<Say voice=\"woman\">" . $part_3 . "</Say>";
            
            $message = $part_1 . $text_space . $part_2 . $text_space . $part_3;
            echo "<Sms>" . $message . "</Sms>";
            
            $results_counter++;
            if ($results_counter == $results_count) break;
        }
        
        echo "<Say voice=\"woman\">Thank you for calling, goodbye.</Say>";
    }
?>
</Response>
```

**Comment by pjaudiomv** [comment]
we had another request for this feature from the facebook

**Comment by rdtripp** [comment]
We have 13 meetings a week.  When someone selects find a meeting I need it to give a list of all meetings for the week.

**Comment by radius314** [comment]
Understood, will work on this for the next version.

**Comment by radius314** [comment]
change is in master, review the readme for usage

**Comment by radius314** [comment]
The section is on Custom Query

**Comment by rdtripp** [comment]
Works like a charm.  Thank you!!
Will be presenting this to North Louisiana ASC this afternoon in Bossier city.

**Comment by pjaudiomv** [comment]
Awesome let the Facebook group know how it goes!

**Comment by rdtripp** [comment]
Will get on facebook when I get back late this evening.  Right now I'm tweaking yap, printing meeting lists. writing PR report, etc, etc ............... never do today what you can do tomorrow and tomorrow is here.

**Comment by radius314** [comment]
@pjaudiomv have verified against your island use case?

**Comment by pjaudiomv** [comment]
no i have not, though I will today.

**Comment by radius314** [comment]
released with 2.5.0

**Comment by pjaudiomv** [comment]
What?!? you closed #11 no wayyy! epic.
